crypto/rand/drbg_lib.c

```diff
@@ -238,8 +312,20 @@ int RAND_DRBG_reseed(RAND_DRBG *drbg,
 
     if (!ctr_reseed(drbg, entropy, entropylen, adin, adinlen))
         goto end;
+
     drbg->state = DRBG_READY;
-    drbg->reseed_counter = 1;
+    drbg->generate_counter = 0;
+
+    if (drbg->reseed_time != 0) {
+        /* automatic reseeding is enabled */
+        if (drbg->parent == NULL) {
+            drbg->reseed_time = time(NULL);
+            drbg->reseed_counter++;
+        } else {
+            drbg->reseed_time = drbg->parent->reseed_time;
+            drbg->reseed_counter = drbg->parent->reseed_counter;
```

Comment:
I think we should take drbg->parent's lock before accessing any of it's variables both here and at other places.

Reply:
There was a lengthy discussion with @paulidale and @dot-asm in PR #4414 whith that result hat reading the reseed count does not have to be atomic, see https://github.com/openssl/openssl/pull/4414#issuecomment-332025551 and https://github.com/openssl/openssl/pull/4414#issuecomment-332460299. For the same reasons, I don't see any benefit from taking a lock here.